src/realm/object-store/shared_realm.cpp

```diff
@@ -81,7 +81,7 @@ Realm::Realm(Config config, util::Optional<VersionID> version, std::shared_ptr<_
     , m_scheduler(m_config.scheduler)
 {
     if (!coordinator->get_cached_schema(m_schema, m_schema_version, m_schema_transaction_version)) {
-        m_transaction = coordinator->begin_read(version.value_or(VersionID{}));
+        m_transaction = coordinator->begin_read();
```

Comment:
So if we don't specify the transaction here, does that mean the version parameter to this constructor is kinda meaningless?

Reply:
Like could the schema change between reading/caching the schema and opening the transaction at the specified version?

Reply:
This is actually reverting to the previous behaviour… so essentially when you create a realm you can pass a version from which you want to read the schema from. But in the past we have always ignored that. 
To answer your question, the schema is cached right after we read and we set up the transaction version etc.

So in theory if you go through the coordinator you should never be able to construct in parallel the same instance, thus when you begin reading you should never be able to clash with the constructor. 

But I suspect using the version there has created some havoc